enhancements/ingress/custom-route-configuration.md

```diff
@@ -0,0 +1,242 @@
+    // This will usually be an operator service account.
+    ConsumingUsers []string
+    // currentHost is the current name used to by the route.  Routes can have more than one exposed name, even though we
+    // only allow one route.spec.host.
+    CurrentHosts []string
+
+    // conditions are available, degraded, progressing.  This allows consistent reporting back and feedback that is well
+    // structured.  These particular conditions have worked very well in ClusterOperators.
+    Conditions []ConfigCondition
+
+    // relatedObjects allows listing resources which are useful when debugging or inspecting how this is applied.
+    // They may be aggregated into an overal status RelatedObjects to be automatically by oc adm inspect
+    RelatedObjects []ObjectReference
+    
+    
+
+    // This API does not include a mechanism to distribute trust, since the ability to write this resource would then 
+    // allow interception.  Instead, if we need such a mechanism, we can talk about creating a way to allow narrowly scoped
+    // updates to a configmap containing ca-bundle.crt for each CustomizeableRoute.
+    // CurrentCABundle []byte
+}
+```
+
+
+### User Stories
+
+#### Cluster-admin wants to customize a route
+To use this, a cluster-admin would
+1.  Either read docs to find the namespace,name tuples or look at an existing cluster and read ingresses.config.openshift.io.
+2.  Create the serving cert/key pair secret in ns/openshift-config
+3.  Create an entry in `ingresses.config.openshift.io.spec.customizeableRouteSpec` for their customization
+4.  Wait to see the corresponding change in status.
+
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Control loop to manage secret read permissions in openshift-config
+List/watch for individual names has been possible for several releases.
+A control loop in either the ingress operator or the cluster config operator will watch for ingress.spec.customizeableRouteSpec.servingCertKeyPairSecret
+and will create a role/rolebinding for the corresponding ingress.status.customizeableRouteStatus.consumingUsers.
+This will allow an operator (or other binary with that user) to get/list/watch on the particular secret.
+
+This means that the power to mutate ingresses.config.openshift.io and ingresses.config.openshfit.io/status
+will imply the power to read secrets in openshift-config.
+If we decide we do not wish to do this, then it will incumbent upon the cluster-admin to create the role and rolebinding.
+
+#### library-go usage
+Because this will be leveraged by the authentication operator (and possibly the console operator), we can provide a config
+observer that reads the ingresses.spec, handles the specified names, sets up the specific list/watch, and takes care of
+copying the secret so it can be mounted.
+
+It would also be possible (though I'm less sure someone would use it), to provide a route injection option for serving cert/key
```

Comment:
Isn't that the only option if your routes use reencrypt/edge termination?

Reply:
I assume that for passthrough routes, the operator (which has a serviceaccount in `ConsumingUsers`) reads the certificate and configures the operand to use it as the serving certificate.